Refactor backend for TCN compliance

rettetdemdativ opened the pull request:
- Merge upload and download APIs
- Add required functions instead of seemingly random ones
- Remove unnecessary clutter

assert-not-singularity committed:
Add Dockerfile

Commit:
Create functions for creating blueprints with database object

Commit:
Use pymondo instead of flask_pymongo
flask_pymongo was causing problems because it automatically connects to mongoDB on localhost and ignored any URI passed to it. Also it features an older version of pymongo, which is totally unnecessary.

Commit:
Create routes with new blueprint creation functions and pass database to them

Commit:
Remove flask-pymongo as dependency

Commit:
Remove flask-pymongo as dependency and add pymongo

Commit:
Bind flask to 0.0.0.0 as host in Docker container

Commit:
Add docker-compose file

Commit:
Adapt tests to new structure

Commit:
Use environment variables in Docker files

Commit:
Merge branch docker into merge-download

Commit:
Merge branch docker into merge-download

Commit:
Change directory structure

Commit:
Remove app directory

Commit:
Rename persistence and model packages

Commit:
Format with black

Commit:
Move test helper functions out of db module

Commit:
Rename main.py to app.py

Commit:
Add get_cases function

Commit:
Add handler to get cases

Commit:
Move random_time_in_the_past to db module:

rettetdemdativ committed:
Format with black

rettetdemdativ committed:
Fix typing errors

rettetdemdativ committed:
Format with black

rettetdemdativ committed:
Install mongodb in actions pipeline

Commit:
Merge pull request #11 from ito-org/actions-test

Install mongodb in actions pipeline

rettetdemdativ committed:
Try testing with localhost mongodb

rettetdemdativ committed:
Use GitHub secret

rettetdemdativ committed:
Remove unused models and routes

rettetdemdativ committed:
Remove unused models and routes

rettetdemdativ committed:
Remove unused models and routes

rettetdemdativ committed:
Rename ApiError to APIError

rettetdemdativ committed:
Remove unused database functions

rettetdemdativ committed:
Add handlers for /report route and simplify initialization

rettetdemdativ committed:
Remove faulty imports

rettetdemdativ committed:
Apply formatting

rettetdemdativ committed:
Rename report_handler to report

rettetdemdativ committed:
Add provisional test for report handler

rettetdemdativ committed:
Add return type for report function

rettetdemdativ committed:
Update GitHub actions workflow

calmandniceperson commented on tests/v0/test_cases.py:
```diff
@@ -1,32 +1,15 @@
-    prev_count: int = count_cases()
-    n: int = 10
-    cases: list = generate_random_cases(n)
+def test_report_post(client: FlaskClient):
```
I'll add more tests before merging this. This was just a first test to see if everything works. Thanks for the suggestion, though!

Commit:
Create index for reportsig during initialization of DBConnection

Commit:
Return an API error if reportsig already exists

Commit:
Add test for duplicates

Commit:
Do nothing if the reportsig already exists
After considering the consequences of returning an error (clients could check which IDs already exist), we decided to do nothing if a duplicate exists

Commit:
Merge pull request #13 from ito-org/reportsig-duplicates

Handle duplicate reportsigs

Commit:
Apply formatting

Commit:
Ignore type checking for pymongo.errors import

Commit:
Apply formatting

Commit:
Fix test for duplicate reportsigs
